dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventSecurityValidationHook.java

```diff
@@ -0,0 +1,112 @@
+package org.hisp.dhis.tracker.validation.hooks;
+
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramInstance;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.program.ProgramStageInstance;
+import org.hisp.dhis.trackedentity.TrackedEntityInstance;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Event;
+import org.hisp.dhis.tracker.preheat.PreheatHelper;
+import org.hisp.dhis.tracker.report.TrackerErrorCode;
+import org.hisp.dhis.tracker.report.TrackerErrorReport;
+import org.hisp.dhis.tracker.report.ValidationErrorReporter;
+import org.hisp.dhis.user.User;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import static org.hisp.dhis.tracker.report.ValidationErrorReporter.newReport;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@Component
+public class EventSecurityValidationHook
+    extends AbstractTrackerValidationHook
+{
+
+    @Override
+    public int getOrder()
+    {
+        return 301;
+    }
+
+    @Override
+    public List<TrackerErrorReport> validate( TrackerBundle bundle )
+    {
+        ValidationErrorReporter reporter = new ValidationErrorReporter( bundle, this.getClass() );
+        User actingUser = bundle.getPreheat().getUser();
+
+        for ( Event event : bundle.getEvents() )
+        {
+            reporter.increment( event );
+
+            ProgramStageInstance programStageInstance = PreheatHelper
+                .getProgramStageInstance( bundle, event.getEvent() );
+            ProgramStage programStage = PreheatHelper.getProgramStage( bundle, event.getProgramStage() );
+            ProgramInstance programInstance = PreheatHelper.getProgramInstance( bundle, event.getEnrollment() );
+            OrganisationUnit organisationUnit = PreheatHelper.getOrganisationUnit( bundle, event.getOrgUnit() );
```

Comment:
does this check if user has write access to the orgUnit?

Reply:
Added org write access check